src/util/fetch.ts

```diff
@@ -12,7 +13,16 @@ export const fetchAPIData = async (url: string, method: string, body:object , he
             ...headers,
         },
     })
-    const response = await data.json();
-    setCookie('shelf-share-session', JSON.stringify(body) , 1 )
-    return response; 
+    
+    if(response.status.toString().startsWith('4'||'5')){
+      console.log("Error", response.status)
+    }
+    else{
+      const data = await response.json();  
+      setCookie('shelf-share-session', JSON.stringify(body) , 1 )
```

Comment:
This setCookie can be deleted. We are already setting the cookie in signup and login adapters.